src/device-config.coffee

```diff
@@ -76,11 +45,23 @@ module.exports = class DeviceConfig
 				.tapCatch (err) =>
 					@logger.logConfigChange(logValue, { err })
 			setBootConfig: (step) =>
-				@config.get('deviceType')
-				.then (deviceType) =>
+				Promise.all [
+					@config.get('deviceType')
+					@getConfigBackend()
+				]
+				.then ([ deviceType, configBackend ]) =>
```

Comment:
Seems `configBackend` is unused. But do you really need to call `getConfigBackend` here? (If yes, Promise.join is typically used in these cases.)

Reply:
That's actually a bug, forgot to switch out the `deviceType` parameter for the `configBackend` param (which also means that I can drop the `.all` altogether.